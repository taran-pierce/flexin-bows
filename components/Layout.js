import Workout from "./Workout";
import Header from "./Header";
import WorkoutList from './WorkoutList'

const Layout = ( props) => {
  const links = [
    {
      name: 'Better Body',
      href: '/better-body'
    },
    {
      name: 'Advanced Conditioning',
      href: '/advanced-conditioning'
    },
    {
      name: 'Body Building',
      href: '/body-building'
    },
    {
      name: 'Strength Training',
      href: '/strength-training'
    },
    {
      name: 'Aerobic',
      href: '/aerobic'
    },
    {
      name: 'Upper Lower Body',
      href: '/upper-lower-body'
    },
    {
      name: 'Anaerobic',
      href: '/anaerobic'
    }
  ]
  
  if (props.data) {
    return (
      <div className={`main-layout`}>
        <style jsx global>{`
          body {
            background: url('/static/images/body-bg.png') top left repeat;
            margin: 0;
            padding: 0;
            font-family: 'Noto Sans JP', Helvetica, sans-serif;
          }

          .hidden {
            display: none;
          }
        `}</style>
        <Header
          links={links}
          onChange={props.onChange}
          searchValue={props.searchvalue}
        />
        <Workout
          data={props.data}
          activeTab={props.activeTab}
          onWorkoutClick={props.onWorkoutClick}
          onBodyPartClick={props.onBodyPartClick}
          image={props.image}
          onChange={props.onChange}
          showDesc={props.showDesc}
        />
      </div>
    )
  } else {
    return (
      <div className={`main-layout`}>
        <style jsx global>{`
          body {
            background: url('/static/images/body-bg.png') top left repeat;
            margin: 0;
            padding: 0;
            font-family: 'Noto Sans JP', Helvetica, sans-serif;
          }

          .hidden {
            display: none;
          }
        `}</style>
        <Header links={links} />
        <WorkoutList links={links} />
      </div>
    )
  }
}

export default Layout