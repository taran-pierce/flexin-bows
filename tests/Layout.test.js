import React from 'react';
import Layout from '../components/Layout';
import renderer from 'react-test-renderer';

test('Does Layout render correctly?', () => {
  
  const component = renderer.create(
    <Layout />,
  );
  
  // verify component compiles correctly
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  // TODO Layout has two ways to render
  // only one is being tested
});
