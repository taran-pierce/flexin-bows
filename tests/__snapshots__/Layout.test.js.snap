// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does Layout render correctly? 1`] = `
<div
  className="jsx-940123235 main-layout"
>
  <header>
    <nav
      className="jsx-3182432235"
    >
      <ul
        className="jsx-3182432235"
      >
        <li
          className="jsx-3182432235"
        >
          <a
            className="jsx-3182432235"
            href="/"
            onClick={[Function]}
          >
            Flexin Bows with Terry
          </a>
        </li>
        <li
          className="jsx-3182432235 search-box"
        >
          <form
            className="jsx-3182432235"
          >
            <label
              className="jsx-3182432235"
            >
              Search:
              <input
                className="jsx-3182432235"
                type="text"
              />
            </label>
          </form>
        </li>
      </ul>
    </nav>
  </header>
  <div>
    <ul>
      <li>
        <a
          href="/better-body"
          onClick={[Function]}
        >
          Better Body
        </a>
      </li>
      <li>
        <a
          href="/advanced-conditioning"
          onClick={[Function]}
        >
          Advanced Conditioning
        </a>
      </li>
      <li>
        <a
          href="/body-building"
          onClick={[Function]}
        >
          Body Building
        </a>
      </li>
      <li>
        <a
          href="/strength-training"
          onClick={[Function]}
        >
          Strength Training
        </a>
      </li>
      <li>
        <a
          href="/aerobic"
          onClick={[Function]}
        >
          Aerobic
        </a>
      </li>
      <li>
        <a
          href="/upper-lower-body"
          onClick={[Function]}
        >
          Upper Lower Body
        </a>
      </li>
      <li>
        <a
          href="/anaerobic"
          onClick={[Function]}
        >
          Anaerobic
        </a>
      </li>
    </ul>
  </div>
</div>
`;
